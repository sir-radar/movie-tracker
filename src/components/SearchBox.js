import React,{useState} from 'react'

const SearchBox = ({onSearch}) => {
  const [search, setSearch] = useState('');
  const [error, setError] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault()
    const error = validate(search)
    setError(error)
    if (error.length > 0) return;
    console.log(search)
    onSearch(search);
  }

  const validate = (title) => {
    if (title.trim().length === 0) return 'Title is required!';
    return '';
  }
  return (
    <div className="row justify-content-center">
      <div className="col-md-10">
        <form onSubmit={handleSubmit}>
          <div className="input-group mb-3">
            <input type="text" onChange={e => setSearch(e.target.value)} className="form-control" placeholder="Search movie title..."/>
            <div className="input-group-append">
              <button className="btn px-3 btn-background"><i className="fa fa-search text-white"></i></button>
            </div>
          </div>
          <p className='error'>{ error }</p>
        </form>
      </div>
    </div>
  )
}

export default SearchBox