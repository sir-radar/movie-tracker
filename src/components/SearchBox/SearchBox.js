import React,{useState} from 'react'

const SearchBox = ({onSearch, updateSerachQuery}) => {
  const [search, setSearch] = useState('');
  const [error, setError] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault()
    const error = validate(search)
    setError(error)
    if (error.length > 0) return;
    updateSerachQuery(search)
    onSearch(search);
  }

  //validate search 
  const validate = (title) => {
    if (title.trim().length === 0) return 'Type in something!';
    return '';
  }
  
  return (
    <div data-testid="search-box" className="row justify-content-center">
      <div className="col-md-10">
        <form data-testid="search-form" onSubmit={handleSubmit}>
          <div className="input-group mb-3">
            <input data-testid="search-input" type="text" onChange={e => setSearch(e.target.value)} className="form-control" placeholder="Search movie title..."/>
            <div className="input-group-append">
              <button className="btn px-3 btn-background"><i className="fa fa-search text-white"></i></button>
            </div>
          </div>
          <p data-testid="search-error" className='error text-danger'>{ error }</p>
        </form>
      </div>
    </div>
  )
}

export default SearchBox