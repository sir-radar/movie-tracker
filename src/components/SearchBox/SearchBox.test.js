import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import SearchBox from './SearchBox';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"

afterEach(cleanup)

const renderComponent = (props) => {
  const div = document.createElement("div");
  return render(<SearchBox {...props}/>, div)
}

const propsData = {
  onSearch: jest.fn(), 
  updateSerachQuery: jest.fn()
};

it("renders without crashing", () => {
  renderComponent(propsData)
})

it("It should update input value", () => {
  const { queryByTestId } = renderComponent(propsData)
  fireEvent.change(queryByTestId('search-input'), { target: { value: 'hey' } })
  expect(queryByTestId('search-input').value).toBe('hey')
})

it("should display error message if input value is empty on form submission", () => {
  const { queryByTestId } = renderComponent(propsData)
  expect(queryByTestId('search-input')).toHaveTextContent("")
  fireEvent.change(queryByTestId('search-input'), { target: { value: '' } })
  fireEvent.submit(queryByTestId('search-form'))
  expect(propsData.onSearch).not.toHaveBeenCalled()
  expect(propsData.updateSerachQuery).not.toHaveBeenCalled()
  expect(queryByTestId('search-error')).toHaveTextContent("Type in something!")
  
})

it("should call onSearch and updateSerachQuery on form submission if input value is valid", () => {
  const { queryByTestId } = renderComponent(propsData)
  expect(queryByTestId('search-input')).toHaveTextContent("")
  fireEvent.change(queryByTestId('search-input'), { target: { value: 'hey' } })
  expect(queryByTestId('search-input').value).toBe('hey')
  fireEvent.submit(queryByTestId('search-form'))
  expect(propsData.onSearch).toHaveBeenCalled()
  expect(propsData.updateSerachQuery).toHaveBeenCalled()
  expect(queryByTestId('search-error')).toHaveTextContent("")
})


it("matches snapshot", () => {
  const tree = renderer.create(<SearchBox {...propsData}/>).toJSON();
  expect(tree).toMatchSnapshot();
})