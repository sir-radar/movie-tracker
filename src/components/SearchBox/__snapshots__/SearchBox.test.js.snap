// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component matches snapshot 1`] = `
<div
  className="row justify-content-center"
  data-testid="search-box"
>
  <div
    className="col-md-10"
  >
    <form
      data-testid="search-form"
      onSubmit={[Function]}
    >
      <div
        className="input-group mb-3"
      >
        <input
          className="form-control"
          data-testid="search-input"
          onChange={[Function]}
          placeholder="Search movie title..."
          type="text"
        />
        <div
          className="input-group-append"
        >
          <button
            className="btn px-3 btn-background"
          >
            <i
              className="fa fa-search text-white"
            />
          </button>
        </div>
      </div>
      <p
        className="error text-danger"
        data-testid="search-error"
      >
        
      </p>
    </form>
  </div>
</div>
`;
